### PR TITLE
OEL-3187: Upgrade to Drupal 11

### PR DESCRIPTION
**Pull request by drishu**
### Description

Upgrade to Drupal 11.

### Change log

- Changed: CAS api usage
- Removed: D9 compatibility
- Security: Upgraded static code analysis

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ services:
       - DOCUMENT_ROOT=/test/oe_authentication
       - COMPOSERVER=--2
   mysql:
-    image: git.fpfis.tech.ec.europa.eu/fpfis/dependency_proxy/containers/percona:5.7
+    image: git.fpfis.tech.ec.europa.eu/fpfis/dependency_proxy/containers/percona:8.0
     command: --innodb-log-file-size=1G --max_allowed_packet=1G --innodb-buffer-pool-size=512M --wait_timeout=3000 --net_write_timeout=3000 --log_error_verbosity=3
```
Comment (brummbar) [inline .drone.yml:23]: Set this command for improved performances:
```
command: --innodb-log-file-size=1G --max_allowed_packet=1G --innodb-buffer-pool-size=512M --wait_timeout=3000 --net_write_timeout=3000 --log_error_verbosity=3 --skip-log-bin --innodb-flush-log-at-trx-commit=2 --innodb-read-io-threads=16 --innodb-write-io-threads=16
```

File: docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: '2'
```
Comment (nagyad) [inline docker-compose.yml:1]: Let's remove this as well.

File: .drone.yml [hunk 3]
```diff
@@ -75,11 +75,5 @@ pipeline:
...
-    - CORE_VERSION: 10.2.0
-      PHP_VERSION: 8.1
-    - CORE_VERSION: 10.2.0
-      PHP_VERSION: 8.2
-    - CORE_VERSION: 10.3.0
-      PHP_VERSION: 8.1
-    - CORE_VERSION: 10.3.0
-      PHP_VERSION: 8.2
+    - CORE_VERSION: 11
+      PHP_VERSION: 8.3
```
Comment (nagyad) [inline .drone.yml:79]: This is not going to work! We need to keep support of D10 until June 2026. We can't force projects to go D11 all of a sudden. Also CAS 3.0 support D10, we should follow suit.

**Review by nagyad — CHANGES_REQUESTED** [review]

File: .drone.yml [hunk 4]
```diff
@@ -75,11 +75,11 @@ pipeline:
...
-      PHP_VERSION: 8.1
-    - CORE_VERSION: 10.3.0
-      PHP_VERSION: 8.2
+      - CORE_VERSION: 10.3.0
+        PHP_VERSION: 8.1
+      - CORE_VERSION: 10.3.0
+        PHP_VERSION: 8.2
+      - CORE_VERSION: 10.3.0
+        PHP_VERSION: 8.3
+      - CORE_VERSION: 11
```
[the remark this answers is not in the document]
Comment (drishu) [reply .drone.yml]: good question, for now 11.0 will be ok

**Review by sergepavle — DISMISSED** [review]

**Review by nagyad — APPROVED** [review]

**Review by sergepavle — DISMISSED** [review]

File: composer.json [hunk 5]
```diff
@@ -7,19 +7,19 @@
...
-        "drupal/core-composer-scaffold": "^10",
+        "drupal/core-composer-scaffold": "^10.3 || ^11.0",
         "drupal/drupal-extension": "^5.0",
-        "drush/drush": "^12",
+        "drush/drush": "^12 || ^13",
         "mikey179/vfsstream": "^1.6.10",
         "nikic/php-parser": "^4",
         "openeuropa/behat-transformation-context": "^0.2",
-        "openeuropa/code-review": "^2.0.0-alpha6",
+        "openeuropa/code-review-drupal": "dev-EWPP-4665",
```
Comment (brummbar) [inline composer.json:22]: This needs to point to a release.

**Review by brummbar — DISMISSED** [review]

File: composer.json [hunk 6]
```diff
@@ -7,19 +7,19 @@
...
+        "drupal/core-composer-scaffold": "^10.3 || ^11.0",
         "drupal/drupal-extension": "^5.0",
-        "drush/drush": "^12",
+        "drush/drush": "^12 || ^13",
         "mikey179/vfsstream": "^1.6.10",
         "nikic/php-parser": "^4",
         "openeuropa/behat-transformation-context": "^0.2",
-        "openeuropa/code-review": "^2.0.0-alpha6",
+        "openeuropa/code-review-drupal": "^1.0.0-alpha",
         "openeuropa/task-runner-drupal-project-symlink": "^1.0-beta6",
```
Comment (drishu) [inline composer.json:23]: we still need to sort out the task runner:
- this merged https://github.com/openeuropa/task-runner/pull/173/files
- this sorted out https://github.com/openeuropa/task-runner-drupal-project-symlink/blob/1.x/composer.json#L11 (wasn't pointing to a release)
- and then some releases for these 2 packages.

Comment (drishu) [reply composer.json]: done

**Review by sergepavle — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Pull request merged**